src/BKTConfig.ts

```diff
@@ -0,0 +1,63 @@
+import { FetchLike } from './internal/remote/fetch'
+
+const MINIMUM_FLUSH_INTERVAL_MILLIS = 60_000 // 60 seconds
+const DEFAULT_FLUSH_INTERVAL_MILLIS = 60_000 // 60 seconds
+const DEFAULT_MAX_QUEUE_SIZE = 50
+const MINIMUM_POLLING_INTERVAL_MILLIS = 60_000 // 60 seconds
+const DEFAULT_POLLING_INTERVAL_MILLIS = 600_000 // 10 minutes
+
+const isValidUrl = (url: string): boolean => {
+  try {
+    new URL(url)
+    return true
+  } catch (err) {
+    return false
+  }
+}
+
+interface RawBKTConfig {
+  apiKey: string
+  apiEndpoint: string
+  featureTag: string
+  eventsFlushInterval?: number
+  eventsMaxBatchQueueCount?: number
+  pollingInterval?: number
+  appVersion: string
+  storageKeyPrefix?: string
+  fetch: FetchLike
+}
+
+export interface BKTConfig extends RawBKTConfig {
+  eventsFlushInterval: number
+  eventsMaxBatchQueueCount: number
+  pollingInterval: number
+}
+
+export const defineBKTConfig = (config: RawBKTConfig): BKTConfig => {
```

Comment:
instead of creating a builder class I've created this factory function for BKTConfig

Reply:
Thanks for the PR!
Can you provide a sample code of how it will look when initializing the SDK?

Reply:
sure, here's a gist

https://gist.github.com/yshrsmz/dc3dbd0eeee878c584ef45532a74702e

Basically, it's going to look like the below:

```typescript
const config = defineBKTConfig({ ... })
const user = defineBKTUser({ ... })
const timeoutMillis = 5_000

initialize(config, user, timeoutMillis)
    // returns Promise<void>
    .then(() => {
      // initial fetch has completed
      const variation = getBKTClient().stringVariation(featureId, 'default-value')
      // do something with a variation value
    })
    .catch((e) => {
      // error while initializing/initial fetch
    })
```